components/next_steps_view/steps/enter_support_email/enter_support_email.tsx

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useState} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {GlobalState} from 'types/store';
+import {getConfig} from 'mattermost-redux/selectors/entities/general';
+import {patchConfig} from 'mattermost-redux/actions/admin';
+import Input from 'components/input';
+import {StepComponentProps} from 'components/next_steps_view/steps';
+import {DispatchFunc} from 'mattermost-redux/types/actions';
+import {isEmail} from 'mattermost-redux/utils/helpers';
+
+import './enter_support_email.scss';
+
+function EnterSupportEmail(props: StepComponentProps): JSX.Element {
+    const [supportEmail, setSupportEmail] = useState('');
+    const [validationError, setValidationError] = useState('');
+    const [updateError, setUpdateError] = useState('');
+    const {formatMessage} = useIntl();
+    const dispatch: DispatchFunc = useDispatch();
+
+    const sEmail = useSelector((state: GlobalState) => {
+        const config = getConfig(state);
+        return config.SupportEmail ? config.SupportEmail : '';
+    });
+
+    useEffect(() => {
+        setSupportEmail(sEmail);
+    }, [sEmail]);
+
+    const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        setSupportEmail(event.target.value);
+    };
+
+    const onKeyUp = () => {
+        setUpdateError('');
+        validateEmail();
+    };
+
+    const validateEmail = (): boolean => {
+        setValidationError('');
+        const valid = isEmail(supportEmail);
+        if (!valid) {
+            setValidationError(formatMessage({id: 'next_steps_view.enter_support_email_step.invalidEmail', defaultMessage: 'Please enter a valid email.'}));
+            return valid;
+        }
+        return valid;
+    };
+
+    const finishStep = async () => {
+        setUpdateError('');
+        const isValid = validateEmail();
+        if (supportEmail !== '' && isValid) {
```

Comment:
shouldn't the empty email validation be part of the validateEmail check?

Reply:
yes it is. made changes.